src/main/kotlin/platform/bukkit/creator/BukkitTemplate.kt

```diff
@@ -34,29 +36,42 @@ object BukkitTemplate : BaseTemplate() {
     fun applyMainClass(
         project: Project,
         packageName: String,
-        className: String
+        className: String,
+        languageType: LanguageType
     ): String {
         val props = mapOf(
             "PACKAGE" to packageName,
             "CLASS_NAME" to className
         )
 
-        return project.applyTemplate(BUKKIT_MAIN_CLASS_TEMPLATE, props)
+        return project.applyTemplate(
+            when (languageType) {
+                LanguageType.JAVA -> BUKKIT_JAVA_MAIN_CLASS_TEMPLATE
+                LanguageType.KOTLIN -> BUKKIT_KOTLIN_MAIN_CLASS_TEMPLATE
```

Comment:
Wouldn't it be simpler to have a method `MinecraftTemplates.getVariant(BUKKIT_MAIN_CLASS_TEMPLATE, language = languageType)` where `BUKKIT_MAIN_CLASS_TEMPLATE = "Bukkit Main Class"` and it would append `language.fileEnding` ?